### ISSUE TITLE
Netplay problems with River City Ransom (start button presses, de-syncs) or slowdown?

### ISSUE THREAD

**Issue by bbbradsmith** [libretro/libretro-fceumm#489]
I have been playing Kunio-kun games with a friend over netplay, and we've been noticing an issue frequently occurring.

Very often, and I think always during moments of slowdown, the game will suddenly register a start-button press for player 1, and the game will be paused for one of us and not the other. After several seconds Retroarch recognizes the de-sync and will resynchronize the two players to match the host's state.

In _**River City Ransom**_ this happened to us very often during battle, and when playing the sequel _**Downtown Special: Kunio-kun no Jidaigeki da yo Zenin Shuugou!**_ it happened extremely often,  which I think is because it has more slowdown than the original. Player 2 would usually be stuck in the pause menu while the player 1 host continues fighting. Because only player 1 can pause or unpause in these games, player 2 has to wait for the full re-sync every time.

I assume this problem would affect more than these two games, but we haven't yet noticed a similar problem from other NES games. The pausing due to start-button is a symptom, but I'm not sure if that means it's getting a frame of all buttons pressed, or if it's specifically only the start button. Also, these games don't react to the second player's start button, which would hide whether wrong-presses are being delivered the other way.

My only theory is that something to do with an input change happening on a lag frame where input isn't being polled might result in divergent behaviour and/or incorrect buttons reported.

**Comment by negativeExponent** [comment]
does the game pauses when not in netplay mode? does overclock help with the slowdown (and possibly that pausing issue). im curious as to why the game forces a pause button press when not intended to...

**Comment by bbbradsmith** [comment]
I am unable to duplicate the problem in local 2-player mode (not netplay).

Unfortunately for the overclocking I can't test netplay on my own. For some reason I can't successfully host (even with relay), only join, so I will try the overclocking next time I can connect with my friend.

**Comment by bbbradsmith** [comment]
Did some testing with my friend after both updating to the latest stable RetroArch (1.10.0) and FCEUmm core. The problem was still occurring. He uses the linux client, and I use windows 7 (64 bit version).

We observed that the spurious game pause could appear on either side. Sometimes it happened for player 1 (host), starting a divergence where eventually player 2 gets a rollback directly into the menu (without seeing the initial pause event), and sometimes it happened for player 2, where player 1 would continue playing and player 2 is now stuck in the menu unable to unpause until they get a rollback to player 1's host state that had never paused.

Since only player 1 can pause the game by pressing start, it seems that the incorrect player 1 input can occur one either side (host, or player 2) even though player 1 should be in control of their own inputs?

We tried with overclocking (post-render) and played for a while without seeing the issue occur, so that does seem to suggest that slowdown is involved in this problem, though it's possible that we were just lucky during this session.

We also tried the QuickNES core as an alternative and played for a while without seeing the issue, just to try and see if it was a universal problem with the RetroArch program rather than the core.

I tried playing against myself through a relay server to induce extra latency, but I wasn't able to reproduce the issue. I don't know if it's because it's hard for me to play the game in the same way we would as two real players. I'm just kind of mashing buttons on both computers at once to test that.

So... sorry if this is a bit difficult to reproduce. I can't seem to offer an easy or predictable way to make it happen.

In _Downtown Special_ it seems to occur on average once per session of battle with a series of enemies (i.e. find an area where enemies are currently, fight about 10 of them to clear it out). That's the game we were using to test. Sometimes it happened more often, like once every minute or two, and sometimes we got a longer stretch where it doesn't happen at all.
